### PR TITLE
[Enhancement] remove the speed limitation for upload/download #24683

### LINKED ISSUE #24683: [Enhancement] remove the speed limitation for upload/download
Problem:
Two problem which limit the download/upload speed:
1. If we use new authentication framework for S3, the client max connection is 25 by default and is not configurable.
2. backup/restore's download/upload task is limited that be can handle at most 3 task in a single backup/restore task.

Solution:
1. Config max connection by config::object_storage_max_connection.
2. introduce max_upload_task_per_be/max_download_task_per_be to control the max task number for download/upload

### PR DESCRIPTION
**Pull request by srlch**
Fixes #24683

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[1094]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[1094]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

File: be/src/fs/fs_s3.cpp [hunk 1]
```diff
@@ -177,6 +177,7 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const TCloudConfigurati
     if (!aws_cloud_credential.endpoint.empty()) {
         config.endpointOverride = aws_cloud_credential.endpoint;
     }
+    config.maxConnections = config::object_storage_max_connection;
```
Comment (chaoyli) [inline be/src/fs/fs_s3.cpp:180]: Does it have an effect on the data lake reader?

Comment (srlch) [reply be/src/fs/fs_s3.cpp:180]: It has no side effect on the data lake reader.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 8 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|3|00.00%|[1093, 1094, 1095]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|2|3|66.67%|[548]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

**Review by decster — APPROVED** [review]

File: be/src/fs/fs_s3.cpp (hunk 1 above)
Comment (dengliu) [reply be/src/fs/fs_s3.cpp:180]: how can we configure the concurrency for the data sync in share-data mode architecture?
cc @kevincai

Comment (kevincai) [reply be/src/fs/fs_s3.cpp:180]: it is a different io framework for shared data, however most of the configuration items are similar, possibly expose similar configurations in that case.

Comment (dengliu) [reply be/src/fs/fs_s3.cpp:180]: > possibly expose similar configurations in that case.

Are you going to cut a PR for that? We want to harness the most from the AWS network bandwidth and S3

Comment (kevincai) [reply be/src/fs/fs_s3.cpp:180]: will take a look. :)

**Review by chaoyli — APPROVED** [review]

**Comment by dengliu** [comment]
thanks for the fix!
can u backport to 3.0?

**Comment by srlch** [comment]
> thanks for the fix!
> can u backport to 3.0?

yes of course, I will backport to 3.0 as soon as possible.

**Comment by yangrong688** [comment]
please also backport this pr to branch 2.5, thanks.

**Comment by srlch** [comment]
> please also backport this pr to branch 2.5, thanks.

2.5 wil also be backport

**Pull request merged**